barretenberg/cpp/src/barretenberg/eccvm/msm_builder.hpp

```diff
@@ -207,146 +207,154 @@ class ECCVMMSMMBuilder {
+                                operation_trace[trace_index] = true;
+                                trace_index++;
+                            }
                             accumulator_trace[msm_row_index] = accumulator;
                             msm_row_index++;
+                        } else {
+                            for (size_t k = 0; k < rows_per_round; ++k) {
+                                auto& row = msm_state[msm_row_index];
+
+                                const size_t points_per_row = (k + 1) * ADDITIONS_PER_ROW > msm_size
+                                                                  ? msm_size % ADDITIONS_PER_ROW
+                                                                  : ADDITIONS_PER_ROW;
+                                const size_t idx = k * ADDITIONS_PER_ROW;
+                                row.msm_transition = false;
+
+                                Element acc_expected = accumulator;
+
+                                for (size_t m = 0; m < 4; ++m) {
+                                    auto& add_state = row.add_state[m];
+                                    add_state.add = points_per_row > m;
+                                    add_state.slice = add_state.add ? msm[idx + m].wnaf_skew ? 7 : 0 : 0;
+
+                                    add_state.point =
+                                        add_state.add
+                                            ? msm[idx + m].precomputed_table[static_cast<size_t>(add_state.slice)]
+                                            : AffineElement{ 0, 0 };
+                                    bool add_predicate = add_state.add ? msm[idx + m].wnaf_skew : false;
+                                    auto p1 = accumulator;
+                                    accumulator = add_predicate ? accumulator + add_state.point : accumulator;
+                                    p1_trace[trace_index] = p1;
+                                    p2_trace[trace_index] = add_state.point;
+                                    p3_trace[trace_index] = accumulator;
+                                    operation_trace[trace_index] = false;
+                                    trace_index++;
+                                }
+                                row.q_add = false;
+                                row.q_double = false;
+                                row.q_skew = true;
+                                row.msm_round = static_cast<uint32_t>(j + 1);
+                                row.msm_size = static_cast<uint32_t>(msm_size);
+                                row.msm_count = static_cast<uint32_t>(idx);
+                                row.pc = pc;
+                                accumulator_trace[msm_row_index] = accumulator;
+                                msm_row_index++;
+                            }
                         }
                     }
                 }
-            }
-        });
+            },
+            HACK_TO_FIX_VANILLA);
```

Comment:
Core change. Adding this leads to about 35 lines changing, almost all of which are due to the formatter, when I construct a diff manually, but github renders the change as much larger.